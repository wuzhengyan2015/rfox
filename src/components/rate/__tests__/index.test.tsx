import { mount, shallow, render, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import Rate from '../index'
import Icon from '../../icon';

configure({ adapter: new Adapter() })

const mockRate = <Rate />

describe('Rate tests', () => { 
    it('renders correctly', () => {
        const tree = render(
            <Rate />
        )
        expect(tree).toMatchSnapshot()
    })
    it('default count prop test', () => {
        const wrapper = shallow(mockRate)
        expect(wrapper.find('.rfox-rate__item').length).toBe(5)
    })
    it('count prop test', () => {
        const wrapper = shallow(<Rate count={10} />)
        expect(wrapper.find('.rfox-rate__item').length).toBe(10)
    })
    it('allClear prop test', () => {
        const wrapper = mount(<Rate allClear/>)
        wrapper.find('.rfox-rate__item').at(2).simulate('click')
        expect(wrapper.find('.rfox-rate__item--hover').length).toBe(3)
        wrapper.find('.rfox-rate__item').at(2).simulate('click')
        expect(wrapper.find('.rfox-rate__item--hover').length).toBe(0)
    })
    it('character prop test', () => {
        const wrapper = shallow(mockRate)
        expect(wrapper.find(Icon).at(0).prop('type')).toBe('icon-star-fill')
        const customWrapper = shallow(<Rate character={<Icon type='icon-heart-fill' />}/>)
        expect(customWrapper.find(Icon).at(0).prop('type')).toBe('icon-heart-fill')
    })
    it('className prop test', () => {
        const wrapper = shallow(<Rate className="unit-test" />)
        expect(wrapper.find('.rfox-rate__item').at(0).hasClass('unit-test')).toBe(true)
    })
    it('count prop test', () => {
        const defaultWrapper = shallow(mockRate)
        expect(defaultWrapper.find('.rfox-rate__item').length).toBe(5)
        const wrapper = shallow(<Rate count={10} />)
        expect(wrapper.find('.rfox-rate__item').length).toBe(10)
    })
    it('count prop test', () => {
        const wrapper = shallow(<Rate defaultValue={2} />)
        expect(wrapper.find('.rfox-rate__item--hover').length).toBe(2)
    })
    it('disabled prop test', () => {
        const wrapper = mount(<Rate disabled/>)
        wrapper.find('.rfox-rate__item').at(0).simulate('mouseover')
        expect(wrapper.find('.rfox-rate__item--hover').length).toBe(0)
        wrapper.find('.rfox-rate__item').at(0).simulate('mouseout')
        expect(wrapper.find('.rfox-rate__item--hover').length).toBe(0)
        wrapper.find('.rfox-rate__item').at(2).simulate('click')
        expect(wrapper.find('.rfox-rate__item--hover').length).toBe(0)
    })
    it('value prop test', () => {
        const wrapper = shallow(<Rate value={3}/>)
        expect(wrapper.find('.rfox-rate__item--hover').length).toBe(3)
        const defaultWrapper = shallow(mockRate)
        defaultWrapper.setProps({value: 3})
        expect(defaultWrapper.find('.rfox-rate__item--hover').length).toBe(3)
    })
    it('activeColor prop test', () => {
        const wrapper = shallow(<Rate value={1} activeColor='#ff0000'/>)
        expect(wrapper.find('.rfox-rate__item').at(0).props().style.color).toBe('#ff0000')
    })
    it('onChange prop test', () => {
        const onChange = jest.fn()
        const wrapper = mount(<Rate onChange={onChange}/>)
        wrapper.find('.rfox-rate__item').at(0).simulate('click')
        expect(onChange).toHaveBeenCalledWith(1)
    })
    it('onFocus prop test', () => {
        const onFocus = jest.fn()
        const wrapper = mount(<Rate onFocus={onFocus}/>)
        wrapper.find('.rfox-rate').at(0).simulate('focus')
        expect(onFocus).toHaveBeenCalled()
    })
    it('onBlur prop test', () => {
        const onBlur = jest.fn()
        const wrapper = mount(<Rate onBlur={onBlur}/>)
        wrapper.find('.rfox-rate').at(0).simulate('blur')
        expect(onBlur).toHaveBeenCalled()
    })
    it('onHoverChange prop test', () => {
        const onHoverChange = jest.fn()
        const wrapper = mount(<Rate onHoverChange={onHoverChange}/>)
        wrapper.find('.rfox-rate__item').at(0).simulate('mouseover')
        expect(onHoverChange).toHaveBeenCalledWith(1)
    })
    it('onMouseOut test', () => {
        const defaultWrapper = mount(mockRate)
        defaultWrapper.find('.rfox-rate__item').at(0).simulate('mouseover')
        expect(defaultWrapper.find('.rfox-rate__item--hover').length).toBe(1)
        defaultWrapper.find('.rfox-rate__item').at(0).simulate('mouseout')
        expect(defaultWrapper.find('.rfox-rate__item--hover').length).toBe(0)
    })
})