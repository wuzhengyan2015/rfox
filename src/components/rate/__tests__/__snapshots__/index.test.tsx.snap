// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rate tests renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "rfox-rate",
      "tabindex": "0",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "rfox-rate__list",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "rfox-rate__item",
              "data-index": "0",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "icon iconfont icon-star-fill",
                  "style": "font-size:24px",
                },
                "children": Array [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "style": undefined,
                },
              },
            ],
            "name": "li",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "rfox-rate__item",
                "data-index": "1",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "icon iconfont icon-star-fill",
                    "style": "font-size:24px",
                  },
                  "children": Array [],
                  "name": "i",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                },
              ],
              "name": "li",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "rfox-rate__item",
                  "data-index": "2",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "icon iconfont icon-star-fill",
                      "style": "font-size:24px",
                    },
                    "children": Array [],
                    "name": "i",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "style": undefined,
                    },
                  },
                ],
                "name": "li",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "rfox-rate__item",
                    "data-index": "3",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "icon iconfont icon-star-fill",
                        "style": "font-size:24px",
                      },
                      "children": Array [],
                      "name": "i",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "style": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "style": undefined,
                      },
                    },
                  ],
                  "name": "li",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "class": "rfox-rate__item",
                      "data-index": "4",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "icon iconfont icon-star-fill",
                          "style": "font-size:24px",
                        },
                        "children": Array [],
                        "name": "i",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                          "style": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                          "style": undefined,
                        },
                      },
                    ],
                    "name": "li",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "data-index": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "data-index": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "data-index": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "data-index": undefined,
                  },
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "data-index": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "data-index": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-index": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-index": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-index": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-index": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "rfox-rate__item",
              "data-index": "1",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "icon iconfont icon-star-fill",
                  "style": "font-size:24px",
                },
                "children": Array [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "style": undefined,
                },
              },
            ],
            "name": "li",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "rfox-rate__item",
                "data-index": "2",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "icon iconfont icon-star-fill",
                    "style": "font-size:24px",
                  },
                  "children": Array [],
                  "name": "i",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                },
              ],
              "name": "li",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "rfox-rate__item",
                  "data-index": "3",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "icon iconfont icon-star-fill",
                      "style": "font-size:24px",
                    },
                    "children": Array [],
                    "name": "i",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "style": undefined,
                    },
                  },
                ],
                "name": "li",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "rfox-rate__item",
                    "data-index": "4",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "icon iconfont icon-star-fill",
                        "style": "font-size:24px",
                      },
                      "children": Array [],
                      "name": "i",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "style": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "style": undefined,
                      },
                    },
                  ],
                  "name": "li",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "data-index": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "data-index": undefined,
                  },
                },
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "data-index": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "data-index": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-index": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-index": undefined,
              },
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "rfox-rate__item",
                "data-index": "0",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "icon iconfont icon-star-fill",
                    "style": "font-size:24px",
                  },
                  "children": Array [],
                  "name": "i",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                },
              ],
              "name": "li",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-index": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-index": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-index": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-index": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "rfox-rate__item",
              "data-index": "2",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "icon iconfont icon-star-fill",
                  "style": "font-size:24px",
                },
                "children": Array [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "style": undefined,
                },
              },
            ],
            "name": "li",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "rfox-rate__item",
                "data-index": "3",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "icon iconfont icon-star-fill",
                    "style": "font-size:24px",
                  },
                  "children": Array [],
                  "name": "i",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                },
              ],
              "name": "li",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Object {
                "attribs": Object {
                  "class": "rfox-rate__item",
                  "data-index": "4",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "icon iconfont icon-star-fill",
                      "style": "font-size:24px",
                    },
                    "children": Array [],
                    "name": "i",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "style": undefined,
                    },
                  },
                ],
                "name": "li",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "data-index": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "data-index": undefined,
                },
              },
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-index": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-index": undefined,
              },
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "rfox-rate__item",
                "data-index": "1",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "icon iconfont icon-star-fill",
                    "style": "font-size:24px",
                  },
                  "children": Array [],
                  "name": "i",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                },
              ],
              "name": "li",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "rfox-rate__item",
                  "data-index": "0",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "icon iconfont icon-star-fill",
                      "style": "font-size:24px",
                    },
                    "children": Array [],
                    "name": "i",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "style": undefined,
                    },
                  },
                ],
                "name": "li",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "data-index": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "data-index": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-index": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-index": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-index": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-index": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "rfox-rate__item",
              "data-index": "3",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "icon iconfont icon-star-fill",
                  "style": "font-size:24px",
                },
                "children": Array [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "style": undefined,
                },
              },
            ],
            "name": "li",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Object {
              "attribs": Object {
                "class": "rfox-rate__item",
                "data-index": "4",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "icon iconfont icon-star-fill",
                    "style": "font-size:24px",
                  },
                  "children": Array [],
                  "name": "i",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                },
              ],
              "name": "li",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-index": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-index": undefined,
              },
            },
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "rfox-rate__item",
                "data-index": "2",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "icon iconfont icon-star-fill",
                    "style": "font-size:24px",
                  },
                  "children": Array [],
                  "name": "i",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                },
              ],
              "name": "li",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "rfox-rate__item",
                  "data-index": "1",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "icon iconfont icon-star-fill",
                      "style": "font-size:24px",
                    },
                    "children": Array [],
                    "name": "i",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "style": undefined,
                    },
                  },
                ],
                "name": "li",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "rfox-rate__item",
                    "data-index": "0",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "icon iconfont icon-star-fill",
                        "style": "font-size:24px",
                      },
                      "children": Array [],
                      "name": "i",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "style": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "style": undefined,
                      },
                    },
                  ],
                  "name": "li",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "data-index": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "data-index": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "data-index": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "data-index": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-index": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-index": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-index": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-index": undefined,
            },
          },
          Object {
            "attribs": Object {
              "class": "rfox-rate__item",
              "data-index": "4",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "icon iconfont icon-star-fill",
                  "style": "font-size:24px",
                },
                "children": Array [],
                "name": "i",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "style": undefined,
                },
              },
            ],
            "name": "li",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "rfox-rate__item",
                "data-index": "3",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "icon iconfont icon-star-fill",
                    "style": "font-size:24px",
                  },
                  "children": Array [],
                  "name": "i",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "style": undefined,
                  },
                },
              ],
              "name": "li",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": Object {
                "attribs": Object {
                  "class": "rfox-rate__item",
                  "data-index": "2",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "icon iconfont icon-star-fill",
                      "style": "font-size:24px",
                    },
                    "children": Array [],
                    "name": "i",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "style": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "style": undefined,
                    },
                  },
                ],
                "name": "li",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "rfox-rate__item",
                    "data-index": "1",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "icon iconfont icon-star-fill",
                        "style": "font-size:24px",
                      },
                      "children": Array [],
                      "name": "i",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "style": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "style": undefined,
                      },
                    },
                  ],
                  "name": "li",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "rfox-rate__item",
                      "data-index": "0",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "icon iconfont icon-star-fill",
                          "style": "font-size:24px",
                        },
                        "children": Array [],
                        "name": "i",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                          "style": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                          "style": undefined,
                        },
                      },
                    ],
                    "name": "li",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "data-index": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "data-index": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "data-index": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "data-index": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "data-index": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "data-index": undefined,
                },
              },
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "data-index": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "data-index": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "data-index": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "data-index": undefined,
            },
          },
        ],
        "name": "ul",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "tabindex": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "tabindex": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
